web-common/src/features/dashboards/url-state/url-state.spec.ts

```diff
@@ -0,0 +1,511 @@
+  },
+
+  {
+    title: "Pivot with no preset and has pivot in state",
+    entity: getPivotedPartialDashboard(
+      [AD_BIDS_PUBLISHER_DIMENSION],
+      [V1TimeGrain.TIME_GRAIN_HOUR],
+      [AD_BIDS_IMPRESSIONS_MEASURE],
+      [AD_BIDS_DOMAIN_DIMENSION],
+      [V1TimeGrain.TIME_GRAIN_DAY],
+    ),
+    expectedUrl:
+      "http://localhost/?vw=pivot&p.r=publisher%2Ctime.hour&p.c=domain%2Ctime.day%2Cimpressions",
+  },
+  {
+    title: "Pivot with preset and has pivot in state same as preset",
+    entity: getPivotedPartialDashboard(
+      [AD_BIDS_PUBLISHER_DIMENSION],
+      [V1TimeGrain.TIME_GRAIN_HOUR],
+      [AD_BIDS_IMPRESSIONS_MEASURE],
+      [AD_BIDS_DOMAIN_DIMENSION],
+      [V1TimeGrain.TIME_GRAIN_DAY],
+    ),
+    preset: {
+      view: V1ExploreWebView.EXPLORE_ACTIVE_PAGE_PIVOT,
+      pivotRows: ["publisher", "time.hour"],
+      pivotCols: ["domain", "time.day", "impressions"],
+    },
+    expectedUrl: "http://localhost/",
+  },
+  {
+    title: "Pivot with preset and pivot in state different as preset",
+    entity: getPivotedPartialDashboard(
+      [AD_BIDS_DOMAIN_DIMENSION],
+      [V1TimeGrain.TIME_GRAIN_DAY],
+      [AD_BIDS_IMPRESSIONS_MEASURE],
+      [],
+      [],
+    ),
+    preset: {
+      view: V1ExploreWebView.EXPLORE_ACTIVE_PAGE_PIVOT,
+      pivotRows: ["publisher", "time.hour"],
+      pivotCols: ["domain", "time.day", "impressions"],
+    },
+    expectedUrl: "http://localhost/?p.r=domain%2Ctime.day&p.c=impressions",
+  },
+  {
+    title: "Pivot with preset and no pivot in state different as preset",
+    entity: getPivotedPartialDashboard([], [], [], [], []),
+    preset: {
+      view: V1ExploreWebView.EXPLORE_ACTIVE_PAGE_PIVOT,
+      pivotRows: ["publisher", "time.hour"],
+      pivotCols: ["domain", "time.day", "impressions"],
+    },
+    expectedUrl: "http://localhost/?vw=overview&p.r=&p.c=",
+  },
+];
+
+describe("Human readable URL state", () => {
+  beforeAll(() => {
+    initLocalUserPreferenceStore(AD_BIDS_EXPLORE_NAME);
+  });
+
+  beforeEach(() => {
+    getLocalUserPreferences().updateTimeZone("UTC");
+    localStorage.setItem(
+      `${AD_BIDS_EXPLORE_NAME}-userPreference`,
+      `{"timezone":"UTC"}`,
+    );
+  });
+
+  describe("Should update url state and restore default state on empty params", () => {
+    for (const { title, entity, preset, expectedUrl } of TestCases) {
+      it(title, () => {
+        const url = new URL("http://localhost");
+        const explore: V1ExploreSpec = {
+          ...AD_BIDS_EXPLORE_INIT,
+          ...(preset ? { defaultPreset: preset } : {}),
+        };
+        const basePreset = getBasePreset(explore, {
+          timeZone: "UTC",
+        });
+        const initEntity = getDefaultMetricsExplorerEntity(
+          AD_BIDS_EXPLORE_NAME,
+          AD_BIDS_METRICS_3_MEASURES_DIMENSIONS,
+          explore,
+          AD_BIDS_TIME_RANGE_SUMMARY,
+        );
+        cleanMetricsExplore(initEntity);
```

Comment:
Why is this necessary?

Reply:
Not all data in MetricsExploreEntity is stored into URL. This cleans it up to simplify comparison. I will rename it to be clearer.